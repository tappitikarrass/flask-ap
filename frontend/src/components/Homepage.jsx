import React, { useEffect } from 'react';
import { useCookies } from 'react-cookie';

import '../scss/Homepage.scss';

function Homepage() {
  const [cookies, setCookie] = useCookies(['token', 'username', 'user_id']);

  useEffect(() => {
    if (cookies.token == null) {
      setCookie(
        'token',
        'undefined',
        {
          path: '/',
          sameSite: 'Lax',
        },
      );
      setCookie(
        'username',
        'undefined',
        {
          path: '/',
          sameSite: 'Lax',
        },
      );
      setCookie(
        'user_id',
        'undefined',
        {
          path: '/',
          sameSite: 'Lax',
        },
      );
    }
  });

  return (
    <div id="homepage" className="content">
      <h2>Anime catalogue SPA build with Flask and ReactJS</h2>
    </div>
  );
}

export default Homepage;
