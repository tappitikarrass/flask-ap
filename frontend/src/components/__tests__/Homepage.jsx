import React from "react";
import { HashRouter } from "react-router-dom";
import { create, act } from "react-test-renderer";
import Homepage from '../Homepage';

describe('Homepage', () => {
  it('',() => {
    let component;
    act(() => {
      component = create(
        <HashRouter>
          <Homepage />
        </HashRouter>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
