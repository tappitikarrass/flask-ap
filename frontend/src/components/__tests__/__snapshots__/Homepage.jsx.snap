// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage  1`] = `
<div
  className="content"
  id="homepage"
>
  <h2>
    Anime catalogue SPA build with Flask and ReactJS
  </h2>
</div>
`;
